sympy/stats/joint_rv.py

```diff
@@ -0,0 +1,195 @@
+"""
+Joint Random Variables Module
+
+See Also
+========
+sympy.stats.rv
+sympy.stats.frv
+sympy.stats.crv
+sympy.stats.drv
+"""
+
+from __future__ import print_function, division
+
+__all__ = ['Joint']
+
+from sympy import Basic, Eq, Mul, Lambda
+from sympy.core.compatibility import string_types
+from sympy.stats.rv import (PSpace, RandomDomain, random_symbols,
+        NamedArgsMixin, RandomSymbol, density)
+from sympy.core.symbol import symbols, Dummy, Symbol
+from sympy.utilities.misc import filldedent
+from sympy.sets.sets import FiniteSet
+from sympy.sets.contains import Contains
+from sympy.integrals.integrals import Integral, integrate
+
+class JointDomain(RandomDomain):
+    """
+    Represents the joint domain as a ProductSet.
+    """
+    @property
+    def set(self):
+        rv_sets = [rv.args[1].domain.set for rv in self.symbols]
+        _set = rv_sets[0]
+        for i in rv_sets[1:]:
+            _set = _set*i
+        return _set
+
+    def as_boolean(self):
+        return Contains(self.symbols, self.set)
+
+    def __contains__(self, other):
+        return other in self.set()
+
+class JointPSpace(PSpace):
```

Comment:
Should this be a subclass of `ProductPSpace`?

Reply:
Yes, I'll add it as a subclass.